test/language/expressions/await/await-awaits-thenable-not-callable.js

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 Microsoft, Inc. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Brian Terlson <brian.terlson@microsoft.com>
+es7id: pending
+description: >
+  Await can await any thenable. If the thenable's then is not callable,
+  await evaluates to the thenable
+---*/
+
+async function foo() {
+  var thenable = { then: 42 };
+  var res = await thenable
+  assert.sameValue(res, thenable);
+}
+
+foo().then($DONE, $DONE);
```

Comment:
This relies on Promise semantics, which are not under test. To some extent, there's no avoiding this, but we can mitigate some issues by incrementing a global `callCount` variable after the assertion, and asserting *that* prior to invoking `$DONE`.

Then again, this is ES7, so maybe we have to start taking correct Promise operation for granted...